File: 240204/비오는 날/rainy-day.java
import java.io.*;
import java.util.*;
import java.util.stream.Collectors;

class Weather {
    String date;
    String day;
    String weather;
    public Weather(String date, String day, String weather) {
        this.date = date;
        this.day = day;
        this.weather = weather;
    }
    
    public void print() {
        System.out.println(String.format("%s %s %s", this.date, this.day, this.weather));
    }
}

public class Main {
    public static int inputLength;
    public static List<Weather> weatherCast = new ArrayList<>();
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        inputLength = Integer.parseInt(br.readLine());
        for (int i = 0; i < inputLength; i++) {
            String[] input = br.readLine().split(" ");
            String date = input[0];
            String day = input[1];
            String weather = input[2];
            weatherCast.add(new Weather(date, day, weather));
        }

        List<Weather> rainWeathers = weatherCast.stream().filter(ele -> ele.weather.equals("Rain")).collect(Collectors.toCollection(ArrayList::new));

        int minIdx = 0;
        for (int i = 1; i < rainWeathers.size(); i++) {
            if (rainWeathers.get(minIdx).date.compareTo(rainWeathers.get(i).date) > 0) {
                minIdx = i;
            }
        }

        rainWeathers.get(minIdx).print();
    }
}

File: 240317/비오는 날/rainy-day.py
class Forecast:
    def __init__(self, date, day, weather):
        self.date = date
        self.day = day
        self.weather = weather

length = int(input())
arr = [tuple(input().split()) for _ in range(length)]
forecast = [Forecast(date, day, weather) for date, day, weather in arr]
rainDay = [Forecast(date, day, weather) for date, day, weather in arr if weather == 'Rain']

min_idx = 0
for i in range(len(rainDay)):
    if (rainDay[i].date < rainDay[min_idx].date):
        min_idx = i
    
print(rainDay[min_idx].date, rainDay[min_idx].day, rainDay[min_idx].weather)